config/config_test.go

```diff
@@ -542,3 +543,76 @@ func TestGetWarpQuorum(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateSourceSubnet(t *testing.T) {
+	validSourceCfg := SourceSubnet{
+		APINodeHost:           "http://test.avax.network",
+		APINodePort:           0,
+		BlockchainID:          testBlockchainID,
+		SubnetID:              testSubnetID,
+		VM:                    "evm",
+		EncryptConnection:     false,
+		SupportedDestinations: []string{testBlockchainID},
+		MessageContracts: map[string]MessageProtocolConfig{
+			testAddress: {
+				MessageFormat: TELEPORTER.String(),
+			},
+		},
+	}
+	testCases := []struct {
+		name                          string
+		sourceSubnet                  func() SourceSubnet
+		destinationBlockchainIDs      []string
+		expectError                   bool
+		expectedSupportedDestinations []string
+	}{
+		{
+			name:                          "valid source subnet; explicitly supported destination",
+			sourceSubnet:                  func() SourceSubnet { return validSourceCfg },
+			destinationBlockchainIDs:      []string{testBlockchainID},
+			expectError:                   false,
+			expectedSupportedDestinations: []string{testBlockchainID},
+		},
+		{
+			name: "valid source subnet; implicitly supported destination",
+			sourceSubnet: func() SourceSubnet {
+				cfg := validSourceCfg
+				cfg.SupportedDestinations = nil
+				return cfg
+			},
+			destinationBlockchainIDs:      []string{testBlockchainID},
+			expectError:                   false,
+			expectedSupportedDestinations: []string{testBlockchainID},
+		},
+		{
+			name:                          "valid source subnet; partially supported destinations",
+			sourceSubnet:                  func() SourceSubnet { return validSourceCfg },
+			destinationBlockchainIDs:      []string{testBlockchainID, testBlockchainID2},
+			expectError:                   false,
+			expectedSupportedDestinations: []string{testBlockchainID},
+		},
+		{
+			name:                          "valid source subnet; unsupported destinations",
+			sourceSubnet:                  func() SourceSubnet { return validSourceCfg },
+			destinationBlockchainIDs:      []string{testBlockchainID2},
+			expectError:                   true,
+			expectedSupportedDestinations: []string{},
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			ids := set.NewSet[string](len(testCase.destinationBlockchainIDs))
+			for _, id := range testCase.destinationBlockchainIDs {
+				ids.Add(id)
+			}
+
+			sourceSubnet := testCase.sourceSubnet()
+			res := sourceSubnet.Validate(&ids)
+			if testCase.expectError {
+				require.Error(t, res)
+			} else {
+				require.NoError(t, res)
+			}
```

Comment:
I think we're missing a check on the `expectedSupportedDestinations` here

Reply:
Added